src/esn.calendar.libs/app/services/fullcalendar/planning-render-event.service.js

```diff
@@ -46,6 +48,27 @@ function calFullCalendarPlanningRenderEventService(
 
         tr.insertBefore(dot, tds[0]);
       }
+
+      function addStyleForEvent(calendar, element, event) {
+        calendar.getOwner().then(function(owner) {
+          var userAsAttendee = calEventUtils.getUserAttendee(event, owner);
+
+          if (userAsAttendee) {
+            if (userAsAttendee.partstat === 'NEEDS-ACTION') {
```

Comment:
you don't need this check since you are dealing with it in the final `else`.

also as huy mentioned, you can use a `switch` statement and add the `element.addClass('planning-event-needs-action');` in the `default` section